pkg/user/tx_client.go

```diff
@@ -293,17 +309,45 @@ func (client *TxClient) BroadcastPayForBlobWithAccount(ctx context.Context, acco
 	gasLimit := uint64(float64(types.DefaultEstimateGas(blobSizes)))
 	fee := uint64(math.Ceil(appconsts.DefaultMinGasPrice * float64(gasLimit)))
 	// prepend calculated params, so it can be overwritten in case the user has specified it.
-	opts = append([]TxOption{SetGasLimit(gasLimit), SetFee(fee)}, opts...)
+
+	latestHeight, err := client.getLatestBlockHeight(ctx)
+	if err != nil {
+		return nil, err
+	}
+	timeOutHeight := uint64(latestHeight) + client.ttlNumBlocks
+
+	txTimeoutHeight, err := GetTimeoutHeightFromOptions(client, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Do not override user set timeout height for the transaction
+	if txTimeoutHeight > 0 {
+		timeOutHeight = txTimeoutHeight
+	}
+
+	opts = append([]TxOption{SetGasLimit(gasLimit), SetFee(fee), SetTimeoutHeight(timeOutHeight)}, opts...)
```

Comment:
```suggestion
	latestHeight, err := client.getLatestBlockHeight(ctx)
	if err != nil {
		return nil, err
	}
	timeOutHeight := uint64(latestHeight) + client.ttlNumBlocks

	opts = append([]TxOption{SetGasLimit(gasLimit), SetFee(fee), SetTimeoutHeight(timeOutHeight)}, opts...)
```
Since you're prepending it, if a user has it set, it will override it